babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1583,3 +1583,49 @@ SqlNode SqlSelectTopN(SqlParserPos pos) :
             SqlLiteral.createBoolean(withTies, pos));
     }
 }
+
+SqlNode SqlHexCharStringLiteral() :
+{
+    final NlsString p;
+    final String formatString;
+    final String charSet;
+    CharacterSet charSetEnum = null;
+    final HexCharLiteralFormat format;
+}
+{
+    (
+        <PREFIXED_HEX_STRING_LITERAL>
+        {
+            charSet = SqlParserUtil
+                      .trim(SqlParserUtil.getCharacterSet(token.image), " ")
+                      .toUpperCase();
+            if (charSet.equals("LATIN")) {
+                charSetEnum = CharacterSet.LATIN;
+            } else if (charSet.equals("UNICODE")) {
+                charSetEnum = CharacterSet.UNICODE;
+            } else if (charSet.equals("GRAPHIC")) {
+               charSetEnum = CharacterSet.GRAPHIC;
+            } else if (charSet.equals("KANJISJIS")) {
+              charSetEnum = CharacterSet.KANJISJIS;
+            } else {
+              throw SqlUtil.newContextException(getPos(),
+              RESOURCE.unknownCharacterSet(charSet));
+            }
+        }
+    |
+        <QUOTED_HEX_STRING>
+    )
+    {
+        String[] tokens = token.image.split("'");
+        p = new NlsString(tokens[1], null, null);
```

Comment:
Same suggestion here. Can we just pass in tokens[1] into the constructor of SqlHexCharString and let it do all of this processing there?

Reply:
Refacotring done in https://github.com/googleinterns/calcite/pull/155/commits/eb9c2fd414f98e835b3ef2d007f0093d7df4e3d9 to address this issue.